src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -102,14 +102,14 @@ private String queryContainerSampleFileRawData(List<QCMetricConfiguration> confi
         sql.append("\nCOALESCE(pci.PrecursorId.Id, pci.MoleculePrecursorId.Id) AS PrecursorId,");
 
         sql.append("\nCOALESCE(X.SeriesLabel, COALESCE(pci.PrecursorId.ModifiedSequence,");
```

Comment:
Can we just select X.SeriesLabel here, and then do a null check in getSeriesLabel() to lazily create the string as needed? And also eliminate the need to call buildSeriesLabel() separately?